qupath-core-processing/src/main/java/qupath/opencv/tools/OpenCVTools.java

```diff
@@ -2668,52 +2668,51 @@ public static List<IndexedPixel> getMaskedPixels(Mat mat, Mat mask) {
 	 * @throws IllegalArgumentException when the input and the mask don't have the same dimensions
 	 */
 	public static float[] extractMaskedPixels(Mat input, Mat mask, int channel) {
-		Mat inputChannel = new Mat();
-		opencv_core.extractChannel(input, inputChannel, channel);
+		try (PointerScope scope = new PointerScope()) {
+			Mat inputChannel = new Mat();
```

Comment:
You can avoid creating a `Mat` here, and delay it to the `else` block in case it is not needed.

(The overhead will be small, but may be more than 'normal' for Java since it will involve a trip through the `PointerScope` for allocation/deallocation, and JavaCPP-managed behavior)